### PR TITLE
Bugfix:  Fixes dtype issue described in #583

### PR DESCRIPTION
**Pull request by dnhkng**
Exllama generates logits in torch Half-dtype, but Outlines requires the Float-dtype. 

This small change converts the logits to the required dtype (whatever that might be), solving issue #583.

Tested with Exllama on the example code on the github front page, and #583 is resolved.

### CODE REVIEW

File: outlines/generate/generator.py [hunk 1]
```diff
@@ -284,5 +284,5 @@ def bias_logits(logits: torch.Tensor, allowed_token_ids: List) -> torch.Tensor:
     """
     biased_logits = torch.full(logits.shape, -math.inf, device=logits.device)
```
Comment (rlouf) [inline outlines/generate/generator.py:285]: Thank you for opening a PR! Could we create `biased_logits` with `dtype=logits.dtype` instead? Or better, use [torch.full_like](https://pytorch.org/docs/stable/generated/torch.full_like.html)

Comment (dnhkng) [reply outlines/generate/generator.py]: Ahh ok, if bias_logits can stay in half format, than this is a nicer solution. I wasn't sure what happens upstream.

**Comment by rlouf** [comment]
Thank you so much for fixing the bug!
